app/models/match.rb

```diff
@@ -97,6 +97,13 @@ def has_points?
     points_home.present? && points_away.present?
   end
 
+  def opponent_of(participant)
+    home = team_home_recursive
+    away = team_away_recursive
+    return home if participant == away
+    away if participant == home
```

Comment:
wow, that really looks weird like that. Wouldn't and explicit return here make it more beautiful?